src/driver/alloy.rs

```diff
@@ -0,0 +1,223 @@
+        if let Some(timestamp) = self.timestamp {
+            block_env.timestamp = U256::from(timestamp);
+        }
+        if let Some(gas_limit) = self.gas_limit {
+            block_env.gas_limit = U256::from(gas_limit);
+        }
+        if let Some(difficulty) = self.difficulty {
+            block_env.difficulty = difficulty;
+        }
+        if let Some(base_fee) = self.base_fee {
+            block_env.basefee = U256::from(base_fee);
+        }
+        if let Some(block_number) = self.block_number.as_number() {
+            block_env.number = U256::from(block_number);
+        }
+    }
+}
+
+impl From<EthCallBundle> for BundleBlockFiller {
+    fn from(bundle: EthCallBundle) -> Self {
+        Self {
+            block_number: bundle.state_block_number,
+            timestamp: bundle.timestamp,
+            gas_limit: bundle.gas_limit,
+            difficulty: bundle.difficulty,
+            base_fee: bundle.base_fee,
+        }
+    }
+}
+
+impl<Ext> BundleDriver<Ext> for EthCallBundle {
+    type Error<Db: revm::Database> = BundleError<Db>;
+
+    fn run_bundle<'a, Db: revm::Database + revm::DatabaseCommit>(
+        &mut self,
+        trevm: crate::EvmNeedsTx<'a, Ext, Db>,
+    ) -> DriveBundleResult<'a, Ext, Db, Self> {
+        // 1. Check if the block we're in is valid for this bundle. Both must match
+        if trevm.inner().block().number.to::<u64>() != self.block_number {
+            return Err(trevm.errored(BundleError::BlockNumberMismatch));
+        }
+
+        let bundle_filler = BundleBlockFiller::from(self.clone());
+
+        let run_result = trevm.try_with_block(&bundle_filler, |trevm| {
+            let mut trevm = trevm;
+
+            for raw_tx in self.txs.clone().into_iter() {
+                let tx = TxEnvelope::decode_2718(&mut raw_tx.to_vec().as_slice());
+
+                let tx = match tx {
+                    Ok(tx) => tx,
+                    Err(e) => return Err(trevm.errored(BundleError::TransactionDecodingError(e))),
+                };
+
+                let run_result = trevm.run_tx(&tx);
+
+                match run_result {
+                    // return immediately if errored
+                    Err(e) => {
+                        return Err(e.map_err(|e| BundleError::EVMError { inner: e }));
+                    }
+                    // Accept the state, and move on
+                    Ok(res) => match res.result() {
+                        ExecutionResult::Revert { .. } | ExecutionResult::Halt { .. } => {
+                            return Err(res.errored(BundleError::BundleReverted));
+                        }
+                        ExecutionResult::Success { .. } => {
+                            trevm = res.accept_state();
+                        }
+                    },
+                }
+            }
+
+            // return the final state
+            Ok(trevm)
+        });
+
+        match run_result {
```

Comment:
this match block is non-functional

Reply:
it is not—else we'd get a compiler error here. there is not semicolon making this an expression therefore it's a return